### DISCUSSION TITLE
Identify VOIP devices by SIP URI from Contact header rather than IP address

### DISCUSSION THREAD

**Discussion by jaminh** [home-assistant/architecture#1096] [General] [answered]
I would like to propose changing the way VOIP devices are identified. Currently VOIP devices are only identified by the IP address the call came from. One issue this causes is an inability to distinguish between calls coming from a centralized Asterisk server but separate endpoints as discussed here https://community.home-assistant.io/t/devices-per-endpoint-when-using-asterisk/688038 . In place of the IP address I think the device should be identified by the full SIP URI as sent in the Contact header field. Using this value would solve the issue of distinguishing between endpoints using a centralized server such as Asterisk, and also provide the URI that would be necessary for making outbound calls from Home Assistant to the VOIP phone as requested here https://github.com/home-assistant-libs/voip-utils/issues/17.

**Comment by elupus** [comment, marked as answer]
I don't think this need an arch discussion. Implement a proposal. IP as identifier is never good.

**Reply by balloob** [reply]
Agreed. If there is a unique ID, we should use that.
